unity-renderer/Assets/DCLPlugins/ECS7/Systems/VideoPlayerSystem/ECSVideoPlayerSystem.cs

```diff
@@ -0,0 +1,65 @@
+using DCL.ECS7.InternalComponents;
+using System.Linq;
+
+namespace ECSSystems.VideoPlayerSystem
+{
+    public class ECSVideoPlayerSystem
+    {
+        public readonly IInternalECSComponent<InternalVideoPlayer> videoPlayerComponent;
+        public readonly IInternalECSComponent<InternalVideoMaterial> videoMaterialComponent;
+
+
+        public ECSVideoPlayerSystem(
+            IInternalECSComponent<InternalVideoPlayer> videoPlayerComponent,
+            IInternalECSComponent<InternalVideoMaterial> videoMaterialComponent)
+        {
+            this.videoPlayerComponent = videoPlayerComponent;
+            this.videoMaterialComponent = videoMaterialComponent;
+        }
+
+        public void Update()
```

Comment:
This approach seems highly inefficient
1. Getting all the playerComponents.
2. Looping through them.
3. Getting all the materials,
4. Loop through them
5. Loop through the textures
6. Get the player for the scene and the videoId
7. Check for all of them IsMaterialAssigned (which relies on LinQ).
8. if material is not assigned, assign it.

has any other alternative have evaluated? mimicking ECS in an environment that is not suitable for ECS (or doesnt bring ECS advantages) seems like having the worst of both worlds.